src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -416,7 +417,7 @@ public void CoverageForOneTest(Guid test, IEnumerable<Mutant> mutants)
             Action<RunEventHandler> updateHandler = null,
             int retries = 0)
         {
-            using var eventHandler = new RunEventHandler(_vsTests, _logger, RunnerId);
+            var eventHandler = new RunEventHandler(_vsTests, _logger, RunnerId);
```

Comment:
Mhm is this done because you're worried there's a 'leak' here due to recursion?

Reply:
Yes, rule S2997: _Remove the 'using' statement; it will cause automatic disposal of 'eventHandler'._ 

I'm not totally sure what I'm doing here. I see the `eventHandler` is returned by the method when `!_vsTestFailed || retries > 10`. I wasn't sure of the meaning of returning a disposed event handler. Wouldn't it be a bug?

I offer to : 
- not dispose the event handler when `!_vsTestFailed || retries > 10` and return it.
- otherwise dispose it, then `return RunTestSession(mutantTestsMap, true, runSettings, updateHandler, ++retries);`

Should I revert the change?

Reply:
Cleaning up the VsTestRunner is probably a job in and of itself, let's revert the change and let some other unfortunate person deal with it another time ;)

Reply:
There is a design issue for sure, but the event handler needs to be disposed here.
The design problem is that `RunEventHandler` aggregates event handling ressources that require to be disposed as well as plain data.
In practice, the eventhandling part is only used within the testrunner logic while the data can be used later.
There should be two classes for that, I think this is a legacy decision that never was improved.

Reply:
The changes around disposing the event handler have been reverted.